datalad/support/gitrepo.py

```diff
@@ -2963,6 +2991,11 @@ def get_content_info(self, paths=None, ref=None, untracked='all'):
                     else 'directory' if path.is_dir() else 'file'
             info[path] = inf
 
+        if ref:
+            # cancel batch process
+            _get_link_target.close()
```

Comment:
Shouldn't be in the finally of try to encompass it? An alternative would be to keep those batched processes known to the instance to be reused across calls. Then have them closed in precommit. That is what AnnexRepo is doing at least

Reply:
try/finally is done. We could think about tracking these processes once we have at least one more use case.